x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/StdDev.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.aggregate;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.compute.aggregation.AggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDevDoubleAggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDevIntAggregatorFunctionSupplier;
+import org.elasticsearch.compute.aggregation.StdDevLongAggregatorFunctionSupplier;
+import org.elasticsearch.xpack.esql.EsqlIllegalArgumentException;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.planner.ToAggregator;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+
+public class StdDev extends AggregateFunction implements ToAggregator {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "StdDev", StdDev::new);
+
+    @FunctionInfo(
+        returnType = "double",
+        description = "The standard deviation of a numeric field.",
+        isAggregation = true,
+        examples = {
+            @Example(file = "stats", tag = "stdev"),
+            @Example(
+                description = "The expression can use inline functions. For example, to calculate the standard "
+                    + "deviation of each employee's maximum salary changes, first use `MV_MAX` on each row, "
+                    + "and then use `STD_DEV` on the result",
+                file = "stats",
+                tag = "docsStatsStdDevNestedExpression"
+            ) }
+    )
+    public StdDev(Source source, @Param(name = "number", type = { "double", "integer", "long" }) Expression field) {
```

Comment:
The params claims to only support the numeric types, but there is no `resolveType` function to enforce this. It should be similar to the Avg function: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Avg.java#L57